src/crimson/os/seastore/seastore_types.h

```diff
@@ -175,7 +177,28 @@ constexpr laddr_t L_ADDR_NULL = std::numeric_limits<laddr_t>::max();
 constexpr laddr_t L_ADDR_ROOT = std::numeric_limits<laddr_t>::max() - 1;
 constexpr laddr_t L_ADDR_LBAT = std::numeric_limits<laddr_t>::max() - 2;
 
-using laddr_le_t = ceph_le64;
+struct laddr_le_t {
+private:
+  laddr_t laddr;
+public:
+  laddr_le_t() = default;
+  laddr_le_t& operator=(laddr_t nv){
+    laddr = nv;
+    return *this;
+  }
+
+  laddr_le_t(const laddr_t &addr)
+  : laddr(addr) {}
+
+  operator laddr_t() const {
+    return laddr;
+  }
+};
+inline laddr_le_t init_laddr_le_t(laddr_t addr) {
+  laddr_le_t v;
+  v = addr;
+  return v;
+}
```

Comment:
This isn't actually correct.  laddr_le_t needs to be little endian, but here you are simply wrapping the normal laddr_t which is machine dependent.  I think perhaps all you needed was to use init_ceph_le64 or add init_laddr_le which simply uses init_ceph_le64.

Reply:
fixedKV need a (const  XX  ) operation. Ceph_le64 doesn't provide it. so add laddr_le_t to implement (const   XX）